engine/src/main/java/io/camunda/zeebe/engine/state/distribution/DbDistributionState.java

```diff
@@ -48,4 +50,66 @@ public DbDistributionState(
             distributionKey,
             commandValueAndValueTypeWrapper);
   }
+
+  @Override
+  public void addCommandDistribution(
+      final long distributionKey, final CommandDistributionRecord commandDistributionRecord) {
+    this.distributionKey.wrapLong(distributionKey);
+    commandDistributionRecordColumnFamily.insert(
+        this.distributionKey,
+        new CommandValueAndValueTypeWrapper().wrap(commandDistributionRecord));
+  }
+
+  @Override
+  public void removeCommandDistribution(final long distributionKey) {
+    this.distributionKey.wrapLong(distributionKey);
+    commandDistributionRecordColumnFamily.deleteIfExists(this.distributionKey);
+  }
+
+  @Override
+  public void addPendingDistribution(final long distributionKey, final int partition) {
+    this.distributionKey.wrapLong(distributionKey);
+    partitionKey.wrapInt(partition);
+    pendingDistributionColumnFamily.insert(distributionPartitionKey, DbNil.INSTANCE);
+  }
+
+  @Override
+  public void removePendingDistribution(final long distributionKey, final int partition) {
+    this.distributionKey.wrapLong(distributionKey);
+    partitionKey.wrapInt(partition);
+    pendingDistributionColumnFamily.deleteExisting(distributionPartitionKey);
+  }
+
+  @Override
+  public boolean hasPendingDistribution(final long distributionKey) {
+    this.distributionKey.wrapLong(distributionKey);
+
+    final var hasPending = new MutableBoolean();
+    pendingDistributionColumnFamily.whileEqualPrefix(
+        this.distributionKey,
+        (compositeKey, dbNil) -> {
+          hasPending.set(true);
+          return false;
+        });
+
+    return hasPending.get();
+  }
+
+  @Override
+  public CommandDistributionRecord getCommandDistributionRecord(
+      final long distributionKey, final int partition) {
+    this.distributionKey.wrapLong(distributionKey);
+    partitionKey.wrapInt(partition);
+
+    final var storedDistribution = commandDistributionRecordColumnFamily.get(this.distributionKey);
+
+    if (storedDistribution == null) {
+      return null;
+    }
+
+    return new CommandDistributionRecord()
+        .setPartitionId(partition)
+        .setValueType(storedDistribution.getValueType())
+        .setRecordValue(storedDistribution.getCommandValue());
```

Comment:
🔧 I think it might be better to leave out the `partitionId` at this spot. Just return a CommandDistributionRecord without a partition id. It should be set on it by the part that knows which partition id to use and send it to.

Reply:
👍 That's a good point. I agree, and it doesn't look strange on the interface, i.e. the method returns the record.

🔧 Let's clarify this in the JavaDoc of the implementation, or in the commit message that adds this line.